.github/workflows.src/tests.tpl.yml

```diff
@@ -130,8 +130,50 @@ jobs:
         path: shared-artifacts
         retention-days: 1
 
+  ui-test:
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+    <<- restore_cache() >>
+
+    # Setup UI test dependencies
+
+    - uses: actions/checkout@v4
+      with:
+        repository: geldata/gel-ui
+        path: build/gel-ui
+        fetch-depth: 2
+
+    - name: Set up Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: lts/*
+
+    - name: Set up chrome
+      uses: browser-actions/setup-chrome@v1
```

Comment:
Please use commit hashes instead of tags on *all* third-party actions (i.e everything that is not in `actions/` or `geldata/`).